### PR TITLE
Validate app-engine-java is installed at the run/debug window.

### LINKED ISSUE #931: Local server run isn't checking for app-engine-java in gcloud
Following is logged if app-engine-java isn't installed:

4:24:53 PM Error running App Engine Local Server
           Error running Google App Engine Dev Server.
           Exception occurred : com.google.cloud.tools.appengine.cloudsdk.AppEngineJavaComponentsNotInstalledException
           Exception message : Validation Error: Java App Engine components not installed. Fix by running 'gcloud components install app-engine-java' on command-line.

### PR DESCRIPTION
**Pull request by joaoandremartins**
Produce user friendlier messaging if it's not.

Fixes #931

### CODE REVIEW

File: google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineStandardRunTask.java [hunk 1]
```diff
@@ -51,12 +53,19 @@ public AppEngineStandardRunTask(@NotNull RunConfiguration runConfig, @Nullable S
...
         .async(true)
-        .startListener(startListener);
+        .startListener(startListener)
+        .build();
 
-    CloudSdkAppEngineDevServer devServer = new CloudSdkAppEngineDevServer(sdkBuilder.build());
+    try {
+      sdk.validateAppEngineJavaComponents();
+    } catch (AppEngineJavaComponentsNotInstalledException ex) {
+      return;
```
Comment (aslo) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineStandardRunTask.java]: What will the user see if we just return here? Will some kind of validation error be displayed to them?

Comment (joaoandremartins) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineStandardRunTask.java]: In that case, the process returned by startListener to AppEngineExecutor will be null.
We're now checking for null in CloudSdkStartupPolicy. If null, we say app-engine-java is missing and provide an instruction on how to install it.

**Comment by elharo** [comment]
FYI, this may soon be moot.  in eclipse we're close to just autoinstalling it when it isn't bundled

**Comment by joaoandremartins** [comment]
Still a good check to not explode if, for any reason, app-engine-java isn't installed, I think.

**Comment by joaoandremartins** [comment]
Can I get an approval here when you guys have time? @aslo @patflynn @etanshaul

**Comment by aslo** [comment]
LGTM :shipit:

**Comment by patflynn** [comment]
I don't think we need this code. The next release of Cloud SDK is going to auto-install the java component if it detects that it's trying to run a java app engine module.

**Comment by patflynn** [comment]
We will need to update to the latest version of the plugins lib to benefit from that.

**Comment by joaoandremartins** [comment]
I prefer to have this code. It's always good to check for a null process in the startup policy.

File: google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineStandardRunTask.java (hunk 1 above)
Comment (patflynn) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineStandardRunTask.java]: why not catch this exception at the caller?  Returning null as an undocumented contract for this specific validation error doesn't seem like an obvious API.

Comment (joaoandremartins) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineStandardRunTask.java]: This method actually doesn't return anything. AppEngineExecutor.getProcess() would. I could try and add some Javadoc there?

I feel like the right way to do this with exceptions would be to throw a checked one, which conflicts with the declaration of `execute()`.

**Pull request merged**